src/internal/connector/onedrive/collection.go

```diff
@@ -223,52 +223,65 @@ func (oc *Collection) populateItems(ctx context.Context) {
 			defer wg.Done()
 			defer func() { <-semaphoreCh }()
 
-			// Read the item
 			var (
+				itemName = *item.GetName()
+				itemSize = *item.GetSize()
 				itemInfo details.ItemInfo
-				itemData io.ReadCloser
-				err      error
-			)
-
-			for i := 1; i <= maxRetries; i++ {
-				itemInfo, itemData, err = oc.itemReader(oc.itemClient, item)
-				if err == nil || graph.IsErrTimeout(err) == nil {
-					// retry on Timeout type errors, break otherwise.
-					break
-				}
-
-				if i < maxRetries {
-					time.Sleep(1 * time.Second)
-				}
-			}
-
-			if err != nil {
-				errUpdater(*item.GetId(), err)
-				return
-			}
-
-			var (
-				itemName string
-				itemSize int64
 			)
 
 			switch oc.source {
 			case SharePointSource:
+				itemInfo.SharePoint = sharePointItemInfo(item, *item.GetSize())
 				itemInfo.SharePoint.ParentPath = parentPathString
-				itemName = itemInfo.SharePoint.ItemName
-				itemSize = itemInfo.SharePoint.Size
 			default:
+				itemInfo.OneDrive = oneDriveItemInfo(item, *item.GetSize())
 				itemInfo.OneDrive.ParentPath = parentPathString
-				itemName = itemInfo.OneDrive.ItemName
-				itemSize = itemInfo.OneDrive.Size
 			}
 
+			// Construct a new lazy readCloser to feed to the collection consumer.
+			// This ensures that downloads won't be attempted unless that consumer
+			// attempts to read bytes.  Assumption is that kopia will check things
+			// like file modtimes before attempting to read.
 			itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+				// Read the item
+				var (
+					itemData io.ReadCloser
+					err      error
+				)
+
+				for i := 1; i <= maxRetries; i++ {
+					_, itemData, err = oc.itemReader(oc.itemClient, item)
+					if err == nil || graph.IsErrTimeout(err) == nil {
```

Comment:
>We should check how kopia handles dir entries for items that fail.

closing the loop on this. Checked the kopia code, it does not return a dir entry if an item has failed so that item will not be available in future backups nor will it be taken into account for kopia-assisted incrementals. Expanding a unit test slightly to ensure files that had errors during read do not result in a file that can be read from kopia